host_core/lib/host_core/providers/provider_module.ex

```diff
@@ -41,7 +41,8 @@ defmodule HostCore.Providers.ProviderModule do
       |> Base.encode64()
       |> to_charlist()
 
-    port = Port.open({:spawn, "#{path}"}, [:binary, {:env, [{'WASMCLOUD_HOST_DATA', host_info}]}])
+    port = Port.open({:spawn, "#{path}"}, [:binary])
+    Port.command(port, "#{host_info}\n")
```

Comment:
Worth noting we can still use environment variables for providers down the line if we want to make any other non-sensitive information available